server/service/osquery_utils/queries.go

```diff
@@ -1351,6 +1352,71 @@ var SoftwareOverrideQueries = map[string]DetailQuery{
 			return mainSoftwareResults
 		},
 	},
+	// deb_last_opened_at collects last opened at information from DEB package files on Linux
+	// hosts. Joining this within the main software query is not performant enough to do on the
+	// device, so we do it on the server instead.
+	"deb_last_opened_at": {
+		Query: `
+		SELECT package, MAX(atime) AS last_opened_at
+		FROM deb_package_files 
+		CROSS JOIN file USING (path) 
+		WHERE type = 'regular' AND regex_match(file.mode, '[1357]', 0)
+		GROUP BY package
+	`,
+		Description:            "A software override query[^1] to append last_opened_at information to Linux DEB software entries.",
+		Platforms:              fleet.HostLinuxOSs,
+		Discovery:              discoveryTable("deb_package_files"), // Table should ship in osquery 5.19.0: https://github.com/osquery/osquery/pull/8657
+		SoftwareProcessResults: processPackageLastOpenedAt("deb_packages"),
+	},
+	// rpm_last_opened_at collects last opened at information from RPM package files on Linux
+	// hosts. Joining this within the main software query is not performant enough to do on the
+	// device, so we do it on the server instead.
+	"rpm_last_opened_at": {
+		Query: // rpm_package_files has a mode column that allows an optimization by filtering before joining to the file table
```

Comment:
Any particular reason this optimization doesn't exist for `deb_package_files`?

Reply:
Yeah. It's because the `rpm_package_files` table uses metadata that's available in the RPM database to find the file mode, while the `deb_package_files` table looks in a list of installed files that includes only the file path so the mode has to be looked up via the `file` table.